packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/utils.js

```diff
@@ -10,89 +10,85 @@ import { create_derived } from '../../utils.js';
 import is_reference from 'is-reference';
 import { locator } from '../../../../../state.js';
 
+/**
+ * @param {Array<AST.Text | AST.ExpressionTag>} values
+ */
+export function get_states_and_calls(values) {
+	let states = 0;
+	let calls = 0;
+	for (let i = 0; i < values.length; i++) {
+		const node = values[i];
+
+		if (node.type === 'ExpressionTag') {
+			if (node.metadata.expression.has_call) {
+				calls++;
+			}
+			if (node.metadata.expression.has_state) {
+				states++;
+			}
+		}
+	}
+
+	return { states, calls };
+}
+
 /**
  * @param {Array<AST.Text | AST.ExpressionTag>} values
  * @param {(node: SvelteNode, state: any) => any} visit
  * @param {ComponentClientTransformState} state
- * @returns {{ value: Expression, has_state: boolean, has_call: boolean, can_inline: boolean }}
+ * @returns {{ value: Expression, has_state: boolean, has_call: boolean }}
  */
-export function build_template_chunk(values, visit, state) {
+export function build_template_literal(values, visit, state) {
```

Comment:
let's keep this rename, `build_template_literal` is a misleading name since it doesn't always produce a template literal

```suggestion
export function build_template_chunk(values, visit, state) {
```